salt/_pillar/metalk8s_endpoints.py

```diff
@@ -16,11 +16,53 @@ def __virtual__():
         return __virtualname__
 
 
-def service_endpoints(service, kubeconfig):
+def namespaced_service_endpoints_nodeport(service, namespace, kubeconfig):
```

Comment:
Hum, no let's not do this _workaround_ thing in the ext_pillar. The service IP is still the cluster IP, so we shouldn't get out of our way just to _make things work_. Instead, you should use `grains['workload_plane_ip']` in `addons.ui.deployed`, until we can reach the real service address through some LB/Ingress.

Reply:
From f2f discussions. This is indeed ugly but we're keeping it for the short duration between here and LB/Ingress addition.